src/viam/sdk/services/mlmodel/client.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Viam Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <viam/api/service/mlmodel/v1/mlmodel.grpc.pb.h>
+
+#include <viam/sdk/services/mlmodel/mlmodel.hpp>
+
+namespace viam {
+namespace sdk {
+
+///
+/// An `MLModelServiceClient` provides client-side access to a
+/// remotely served ML Model Service. Use this class to communicate
+/// with MLModelService instances running elsewhere.
+///
+class MLModelServiceClient : public MLModelService {
+   public:
+    using service_type = viam::service::mlmodel::v1::MLModelService;
+
+    MLModelServiceClient(std::string name, std::shared_ptr<grpc::Channel> channel);
+
+    std::shared_ptr<named_tensor_views> infer(const named_tensor_views& inputs) override;
+    struct metadata metadata() override;
+
+   protected:
+    MLModelServiceClient(std::string name, std::unique_ptr<service_type::StubInterface> stub);
```

Comment:
[q] Where is this used? This reminds me of one of the old testing methods

Reply:
It isn't. leftovers from before, and I've now removed it.

Reply:
Oops, no. I take that back. It *is* used. I'll drop a comment on where.